cmd/datamon/cmd/flags.go

```diff
@@ -432,10 +395,131 @@ func paramsToDestStore(params flagsT,
 	return destStore, nil
 }
 
-func paramsToContributor(_ flagsT) (model.Contributor, error) {
-	return authorizer.Principal(config.Credential)
+/** combined config (file + env var) and parameters (pflags) */
+
+type cliOptionInputs struct {
+	config *CLIConfig
+	params *flagsT
+}
+
+func newCliOptionInputs(config *CLIConfig, params *flagsT) *cliOptionInputs {
+	return &cliOptionInputs{
+		config: config,
+		params: params,
+	}
+}
+
+/** combined config and parameters to internal objects */
+
+func (in *cliOptionInputs) datamonContext(ctx context.Context) (context2.Stores, error) {
+	// here we select a 100% gcs backend strategy (more elaborate strategies could be defined by the context pkg)
+	return gcscontext.MakeContext(ctx,
+		in.params.context.Descriptor,
+		in.config.Credential,
+		gcs.Logger(in.mustGetLogger()))
+}
+
+func (in *cliOptionInputs) srcStore(ctx context.Context, create bool) (storage.Store, error) {
+	var (
+		err                 error
+		consumableStorePath string
+	)
+	switch {
+	case in.params.bundle.DataPath == "":
+		consumableStorePath, err = ioutil.TempDir("", "datamon-mount-destination")
+		if err != nil {
+			return nil, fmt.Errorf("couldn't create temporary directory: %w", err)
+		}
+	case strings.HasPrefix(in.params.bundle.DataPath, "gs://"):
+		consumableStorePath = in.params.bundle.DataPath
+	default:
+		consumableStorePath, err = sanitizePath(in.params.bundle.DataPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to sanitize destination: %v: %w",
+				in.params.bundle.DataPath, err)
+		}
+	}
+
+	if create {
+		createPath(consumableStorePath)
+	}
+
+	var sourceStore storage.Store
+	if strings.HasPrefix(consumableStorePath, "gs://") {
+		fmt.Println(consumableStorePath[4:])
+		sourceStore, err = gcs.New(ctx,
+			consumableStorePath[5:],
+			in.config.Credential,
+			gcs.Logger(in.mustGetLogger()))
+		if err != nil {
+			return sourceStore, err
+		}
+	} else {
+		DieIfNotAccessible(consumableStorePath)
+		DieIfNotDirectory(consumableStorePath)
+		sourceStore = localfs.New(afero.NewBasePathFs(afero.NewOsFs(), consumableStorePath))
+	}
+	return sourceStore, nil
+}
+
+func (in *cliOptionInputs) bundleOpts(ctx context.Context) ([]core.BundleOption, error) {
+	stores, err := in.datamonContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ops := []core.BundleOption{
+		core.ContextStores(stores),
+	}
+	return ops, nil
 }
 
+func (in *cliOptionInputs) mustGetLogger() *zap.Logger {
+	in.config.onceLogger.Do(func() {
+		var err error
+		in.config.logger, err = dlogger.GetLogger(in.params.root.logLevel)
+		if err != nil {
+			wrapFatalln("failed to set log level", err)
```

Comment:
np removing `wrapFatalln` calls from `cliOptionInputs` since it's straightforward, not a lot of further decision-making to do here.

splitting off another package, which totally needs to happen eventually, is probably going to involve more considered decisions than this patch ever involved, so i'll defer that part until later.

Reply:
ok good